src/main/java/com/getbase/sync/SyncService.java

```diff
@@ -29,8 +34,20 @@ public Session start(String deviceUUID) {
                 buildHeaders(deviceUUID),
                 null);
 
-        if (response.getHttpStatus() == 204) return null;
-        return SessionDeserializer.deserialize(JsonDeserializer.deserializeRaw(response.getBody()));
+        if (response.getHttpStatus() == 204) {
+            log.info("Received 204 from sync start. Nothing to fetch.");
+            return null;
+        }
+
+        final Session session = SessionDeserializer.deserialize(JsonDeserializer.deserializeRaw(response.getBody()));
+
+        if (session != null) {
+            log.info("Started sync process with pending queues: {}", session.getQueues());
+        } else {
+            log.warn("Sync process will not be started due to missing session data: {}", response);
```

Comment:
Isn't it an exceptional conditional that shall be handled by throwing exception?